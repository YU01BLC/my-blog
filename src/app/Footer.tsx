import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import React from 'react';

export default function Footer() {
  return (
    <Box sx={{ position: 'absolute', bottom: 0, width: '100%', borderTop: 'solid', padding: '1em' }}>
      <Typography>@2023 shincode</Typography>
    </Box>
  );
}
