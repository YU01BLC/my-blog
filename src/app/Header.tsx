import React from 'react';
import Image from 'next/image';
import AppBar from '@mui/material/AppBar';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import Box from '@mui/material/Box';
import Button from '@mui/material/Button';
import Link from 'next/link';
import BlogImage from '../../public/assets/blogIcon.png';

export default function Header() {
  return (
    <AppBar position='relative'>
      <Toolbar sx={{ bgcolor: '#37474f', borderBottom: 'solid' }}>
        <Image src={BlogImage} alt='blog icon' style={{ width: '32px', height: '32px', marginRight: '1em' }} />
        <Box
          sx={{
            display: 'flex',
            position: 'relative',
            justifyContent: 'space-between',
            width: '100%',
          }}
        >
          <Typography variant='h6' noWrap>
            <Link href='/'>My Blog</Link>
          </Typography>
          <Typography variant='h6' color='inherit' noWrap>
            <Button variant='contained' color='warning'>
              <Link href='/articles'>記事を書く</Link>
            </Button>
          </Typography>
        </Box>
      </Toolbar>
    </AppBar>
  );
}
