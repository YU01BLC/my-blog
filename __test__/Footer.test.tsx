import { render, screen } from "@testing-library/react";
import Header from "../src/app/Header";
import Footer from "../src/app/Footer";
import Page from "../src/app/page";
import userEvent from '@testing-library/user-event'
import "@testing-library/jest-dom";

describe("Page(Footer)コンポーネントSnapshot", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
});

// Footerコンポーネント画面表示要素確認テスト
describe("Page(Footer)コンポーネント表示確認", () => {
   it("テキスト「@2023 shincode」が表示されること", () => {
    render(<Footer />);
    const headElement = screen.getByText("@2023 shincode");
    expect(headElement).toBeInTheDocument();
  });
});

// ページ遷移テスト
describe("ページ遷移テスト", () => {
  // app routerでのページ遷移に関するテスト方法が現状不明。一旦スキップで対応する。
  xit("テキスト「記事を書く」押下時に「Articles」コンポーネント要素のテキスト「articles」が表示されること", async () => {
    render(<Header />);
    userEvent.click(screen.getByText("記事を書く"))
    expect(await screen.getByText('articles')).toBeInTheDocument()
  });
  xit("テキスト「My Blog」押下時に「MyBlog(最上位Page)」コンポーネント要素のテキスト「記事投稿」が表示されること", async () => {
    render(<Page />);
    userEvent.click(screen.getByText("My Blog"))
    expect(await screen.getByText('記事投稿')).toBeInTheDocument()
  });
});