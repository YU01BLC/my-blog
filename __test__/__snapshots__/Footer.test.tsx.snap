// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page(Footer)コンポーネントSnapshot snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1e6w83v"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
    >
      @2023 shincode
    </p>
  </div>
</DocumentFragment>
`;
